include/fast_float/float_common.h

```diff
@@ -89,11 +102,23 @@ inline bool fastfloat_strncasecmp(const char *input1, const char *input2,
 #error "FLT_EVAL_METHOD should be defined, please include cfloat."
 #endif
 
-namespace {
-constexpr uint32_t max_digits = 768;
-constexpr uint32_t max_digit_without_overflow = 19;
-constexpr int32_t decimal_point_range = 2047;
-} // namespace
+// a pointer and a length to a contiguous block of memory
+template <typename T>
+struct slice {
```

Comment:
This is fine but there is already a standard std::slice type, see https://www.cplusplus.com/reference/valarray/slice/slice/

I'd like to have your thoughts on this... are we concerned that the name overload could be confusing?

Furthermore, why should it be a template if it is always templated with the same type? Generality is great, when it is actually put into good use, but if we never use the extra freedom, then it is not so beneficial.

Reply:
I'll rename it. There's numerous other names possible. And I do actually use it for one other template: it's for both a [limb_slice](https://github.com/Alexhuszagh/fast_float/blob/6758fe129dd1354bea8039f06c6cd8dde7335651/include/fast_float/bigint.h#L29) and [byte_slice](https://github.com/Alexhuszagh/fast_float/blob/6758fe129dd1354bea8039f06c6cd8dde7335651/include/fast_float/ascii_number.h#L72)

Reply:
> And I do actually use it for one other template: it's for both a limb_slice and byte_slice

Yes. I noticed that later.

Reply:
Does `span` sound good? C++20 has the same concept, and uses the same name for it. They're implemented practically the same way, as a pointer and a size. It would basically be minimal backport for C++11, essentially. If not, I can think of another name.

Reply:
Span seems much better and less confusing.


I am less concerned about us using the same name, but more about the conceptual confusion. When you read code, it helps if the same name refers to the same usage. Otherwise it makes the code harder to read than it needs to.

(I got stuck on your code because I was assuming that you had used std::slice... and I could not understand the code at first.)